dw-popover-dialog.js

```diff
@@ -0,0 +1,312 @@
+
+
+/**
+ * 
+ * Purpose: 
+ *  - Renders content into popover.
+ * 
+ * Usage: 
+ *  - This element can be used by extention only. e.g `<my-popover-dialog></my-popover-dialog>`
+ *  - CSS properties:
+ *    - `--dw-popover-min-width`: Minimum width of popover dialog. Default is 280px
+ *    - `--dw-popover-width`: Width of popover dialog. Default is 280px.
+ *    - `--dw-popover-overlay-background`: Background of popover overlay. Default is `rgba(0, 0, 0, 0.3)`
+ *    - `--dw-popover-animation-time`: Animation time of popover. Default is 0.3s
+ *  - Events:
+ *    - Dispatches `dw-dialog-opened` & `dw-dialog-closed` events.
+ * Implementaiton Notes:
+ *  - Extends `DwComposeBaseDialogMixin`
+ *  - Renders popover using tippy.js
+ *
+ */
+export const DwPopoverDialogMixin = (baseElement) => class DwPopoverDialog extends DwCompositeBaseDialogMixin(baseElement) {
+  static get styles() {
+    return [
+      popoverStyle,
+    ];
+  }
+
+  static get properties() {
+    return {
+      /**
+       * Opens dialog when true.
+       * Closes dialog when false
+       */
+      opened: { type: Boolean, reflect: true },
+
+      /**
+       * Trigger element for which `popover` dialog is opened.
+       */
+      triggerElement: { type: Object },
+
+      /**
+       * When it's `true`, it hides `triggerElement` when `popover` dialog is opened.
+       */
+      showTrigger: { type: Boolean },
+
+      /**
+       * It's horizontal & vertical offset from `triggerElement`. Value is in pixel. 
+       * Default is [0, 0]
+       */
+      popoverOffset: { type: Array },
+
+      /**
+       * Animation of popover dialog. 
+       * Possible values are: 'dropdown' or 'scale'. Default is 'dropdown'
+       * Note: To apply custom animation, style should be imported explicitly where trigger element exists.
+ *     * e.g: if popoverAnimation is "fadeIn" then animation styling should be applied on `.tippy-box[data-animation="fadeIn"]` selector.
+       */
+      popoverAnimation: { type: String },
+
+      /**
+       * Sets the placement of the popover dialog
+       * Possible value: `bottom-start` or `bottom-end`
+       */
+      popoverPlacement: { type: String },
+
+      /**
+       * Maximum width of popover dialog. Default is 1000;
+       */
+      popoverMaxWidth: { type: Number },
+
+      /**
+      * Input property. When it's `true`, it's height will be full (up to the bottom of viewport)
+      */
+      fullHeight: { type: Boolean }
+    };
+  }
+
+  get opened() {
+    return this._opened;
+  }
+
+  set opened(val) {
+    const oldVal = this._opened;
+    if (oldVal === val) {
+      return;
+    }
+
+    this._opened = val;
+    this.requestUpdate('opened', oldVal);
+    this._onOpenedChanged(val);
+  }
+
+  render() {
+    if (this.type !== 'popover') {
+      return html`${super.render()}`;
+    }
+
+    return html`
+      <div id="popover_dialog__surface">
```

Comment:
@nirmalbaldaniya-dream अगर user footer template में footer नहीं देता हे and user footer को कोई भी height set करता हे तो इस layout के हिसाब से वो footer के लिए निचे जगह रोके गा, इस use case के लिए template को update करना चाहिए